### DISCUSSION TITLE
Feature request: Watch multiple values to run func

### DISCUSSION THREAD

**Discussion by jordyvandomselaar** [alpinejs/alpine#799] [4. General]
It would be sweet if we could do:

```javascript
this.$watch(['some', 'thing'], (some, thing) => {
    //
})
```

**Comment by HugoDF** [comment]
Could this be implemented as a custom magic property? @KevinBatdorf @ryangjchandler 

Would be worth raising as a dicussion or issue there maybe https://github.com/alpine-collective/alpine-magic-helpers

**Reply by KevinBatdorf** [reply]
I'm not sure we can directly, but @SimoTod mentioned we might be able to just replicate `$watch()` so that we could watch parent components, etc. I'll try something out this weekend.

@jordyvandomselaar If you need this now, you're probably best to just set up a loop to initialize multiple watchers at once.

**Reply by HugoDF** [reply]
Yeah I guess something like:

```js
function component() {
  return {
    init() {
      this.$watchMany(['some', 'thing'], (some, thing) => {});
    },
    $watchMany(fields, handler) {
      fields.forEach((field, idx) => {
        this.$watch(field, (val) => {
          const update = fields.map((f) => f === field ? val : this[f]))
          handler(...update);
        });
      }); 
    }
  }
}
```

**Reply by SimoTod** [reply]
Yeah, from an helper, I'm pretty confident we can access $el.__x unobservedData.$watch() @KevinBatdorf

**Reply by jordyvandomselaar** [reply]
> I'm not sure we can directly, but @SimoTod mentioned we might be able to just replicate `$watch()` so that we could watch parent components, etc. I'll try something out this weekend.
> 
> @jordyvandomselaar If you need this now, you're probably best to just set up a loop to initialize multiple watchers at once.

That's what I'm currently doing, thanks all for looking into it! =)
